src/LndClient/Data/Newtype.hs

```diff
@@ -74,23 +74,39 @@ newtype RawTx = RawTx {unRawTx :: ByteString}
 
 instance Out RawTx
 
-newtype ChanId = ChanId {unChanId :: Word64}
+newtype ChanId = ChanId {unChanId :: Natural}
   deriving newtype
-    ( PersistField,
-      PersistFieldSql,
-      Eq,
+    ( Eq,
       Ord,
       Show,
       Read
     )
   deriving stock (Generic)
 
+instance PersistField ChanId where
+  toPersistValue =
+    toPersistValue . unsafeFrom @Natural @Word64 . unChanId
+  fromPersistValue =
+    (ChanId . from @Word32 @Natural <$>)
```

Comment:
For me is not clear why it's Word32 but not Word64 here. Probably a typo, please fix.